### PR TITLE
add BME280 to Staging

### PR DESCRIPTION
**Pull request by magallant**

### CODE REVIEW

File: libs/pilight/protocols/GPIO/bme280.c [hunk 1]
```diff
@@ -0,0 +1,540 @@
+/*
+	Copyright (C) 2014 CurlyMo & hstroh
+
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.c:3]: Can you change this in MPLv2?

File: libs/pilight/protocols/GPIO/bme280.c [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+	bme280data->dig_p1 = 0;
+	bme280data->dig_p2 = 0;
+	bme280data->dig_p3 = 0;
+        bme280data->dig_p4 = 0;
+        bme280data->dig_p5 = 0;
+        bme280data->dig_p6 = 0;
+        bme280data->dig_p7 = 0;
+        bme280data->dig_p8 = 0;
+        bme280data->dig_p9 = 0;
+	bme280data->dig_h1 = 0;
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.c:108]: Indenting is off

File: libs/pilight/protocols/GPIO/bme280.c [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+        }
+        pre_oversampling &= 0x7;
+
+	// resize the memory blocks pointed to by the different pointers
+	size_t sz = (size_t) (bme280data->nrid + 1);
+	unsigned long int sizeUChar = sizeof(uint8_t) * sz;
+        unsigned long int sizeChar = sizeof(int8_t) * sz;
+	unsigned long int sizeShort = sizeof(int32_t) * sz;
+	unsigned long int sizeUShort = sizeof(uint32_t) * sz;
+	bme280data->fd = REALLOC(bme280data->fd, (sizeof(int) * sz));
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.c:163]: Can you error check if these succeeded?

File: libs/pilight/protocols/GPIO/bme280.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*
+	Copyright (C) 2014 CurlyMo & hstroh
+
+	This file is part of pilight.
+
+	pilight is free software: you can redistribute it and/or modify it under the
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.h]: MPLv2

File: libs/pilight/protocols/GPIO/bme280.c [hunk 5]
```diff
@@ -1,19 +1,8 @@
...
-	Foundation, either version 3 of the License, or (at your option) any later
-	version.
-
-	pilight is distributed in the hope that it will be useful, but WITHOUT ANY
-	WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
-	A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
-
-	You should have received a copy of the GNU General Public License
-	along with pilight. If not, see	<http://www.gnu.org/licenses/>
+	Copyright (C) CurlyMo
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.c]: Missing newline, and can you keep yourself in the copyright?

File: libs/pilight/protocols/GPIO/bme280.c [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+
+	if(json_find_number(json, "poll-interval", &itmp) == 0)
+		interval = (int) round(itmp);
+	json_find_number(json, "temperature-offset", &temp_offset);
+	json_find_number(json, "pressure-offset", &pressure_offset);
+	if(json_find_number(json, "hum-oversampling", &itmp) == 0) {
+		hum_oversampling = (uint8_t) itmp;
+	}
+	hum_oversampling &= 0x7;
+        if(json_find_number(json, "tem-oversampling", &itmp) == 0) {
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.c]: Many more indenting is off

File: libs/pilight/protocols/GPIO/bme280.h [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*
-	Copyright (C) CurlyMo
+	Copyright (C) CurlyMo and MaGallant
 	This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.h]: Newline

File: libs/pilight/protocols/GPIO/bme280.c [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*
-	Copyright (C) CurlyMo
+	Copyright (C) CurlyMo and MaGallant
 	This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (CurlyMoo) [inline libs/pilight/protocols/GPIO/bme280.c]: Newline

File: libs/pilight/protocols/GPIO/bme280.c (hunk 3 above)
Comment (magallant) [reply libs/pilight/protocols/GPIO/bme280.c:163]: Isn't that done in the following ? :
`        bme280data->dig_h6 = REALLOC(bme280data->dig_h6, sizeChar);

        if(bme280data->dig_t1 == NULL || bme280data->dig_t2 == NULL || bme280data->dig_t3 == NULL || bme280data->dig_p1 == NULL ||
                bme280data->dig_p2 == NULL || bme280data->dig_p3 == NULL || bme280data->dig_p4 == NULL || bme280data->dig_p5 == NULL $
                bme280data->dig_p6 == NULL || bme280data->dig_p7 == NULL || bme280data->dig_p8 == NULL || bme280data->dig_p9 == NULL $
                bme280data->dig_h1 == NULL || bme280data->dig_h2 == NULL || bme280data->dig_h3 == NULL || bme280data->dig_h4 == NULL $
                bme280data->dig_h5 == NULL || bme280data->dig_h6 == NULL || bme280data->fd == NULL) {
                fprintf(stderr, "out of memory\n");
                exit(EXIT_FAILURE);
        }
`

Comment (CurlyMoo) [reply libs/pilight/protocols/GPIO/bme280.c:163]: Yes, missed that.

**Comment by CurlyMoo** [comment]
While i will continue reviewing this code, can you also make a additional PR to the rewrite branch with a unittest.

File: libs/pilight/protocols/GPIO/bme280.c (hunk 6 above)
Comment (magallant) [reply libs/pilight/protocols/GPIO/bme280.c]: I hope I nail them all now ...

**Pull request merged**